e2e/testcafe-devextreme/tests/editors/list/focus.ts

```diff
@@ -7,13 +7,54 @@ import { createWidget } from '../../../helpers/createWidget';
 fixture`List`
   .page(url(__dirname, '../../container.html'));
 
+const LIST_ITEM_DELETE_BUTTON = 'dx-list-static-delete-button';
+
 const createList = (selectionMode, allowItemDeleting = false) => createWidget('dxList', {
   items: ['item1', 'item2', 'item3'],
   showSelectionControls: true,
   selectionMode,
   allowItemDeleting,
 });
 
+test('Should not focus item when deleting when focusStateEnabled = false (T1226030)', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+  const list = new List('#container');
+  await list.option({
+    focusStateEnabled: false,
+    itemDeleteMode: 'static',
+  });
+  const firstItem = list.getItem(0);
+  const $firstDeleteBtn = firstItem.element.find(`.${LIST_ITEM_DELETE_BUTTON}`);
+
+  await t
+    .click($firstDeleteBtn)
+    .expect(firstItem.isFocused)
+    .notOk();
+  await testScreenshot(t, takeScreenshot, 'First item should not be focused when deleted.png', { element: '#container' });
+  await t
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => createList('none', true));
+
+test('Should focus item when deleting when focusStateEnabled = true (T1226030)', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+  const list = new List('#container');
+  await list.option({
+    itemDeleteMode: 'static',
+  });
+  const firstItem = list.getItem(0);
+  const $firstDeleteBtn = firstItem.element.find(`.${LIST_ITEM_DELETE_BUTTON}`);
+
+  await t
+    .click($firstDeleteBtn)
+    .expect(firstItem.isFocused)
+    .ok();
+  await testScreenshot(t, takeScreenshot, 'First item should be focused when deleted.png', { element: '#container' });
+  await t
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => createList('none', true));
```

Comment:
Actually not a UI is important for us here but just a fact that item is focused. It can be checked by QUnit test through `item.isFocused` as it was done in the [related PR](https://github.com/DevExpress/DevExtreme/pull/25949/files).

Screenshot testing should only be used for UI testing or some complex scenario which it's difficult to emulate.

Reply:
Since you have already written a testcafe test, let's not delete it, just add an additional QUnit test so that in the future we can quickly see the crash locally by running QUnit test

Reply:
[7985c4b](https://github.com/DevExpress/DevExtreme/pull/27230/commits/7985c4b7aa2403c7df668bb04050c83ff10c4325)